lumicks/pylake/channel.py

```diff
@@ -504,6 +504,172 @@ def plot(self, start=None, **kwargs):
         plt.ylabel(self.labels.get("y", "y"))
         plt.title(self.labels.get("title", "title"))
 
+    @staticmethod
+    def _annotate(start_time, annotation, annotation_direction, stop_time, color):
+        """Annotate a region or time point with some text
+
+        Parameters
+        ----------
+        start_time : float
+            Start time in seconds
+        annotation : str | None
+            String to annotate with
+        annotation_direction : "horizontal" | "vertical" | None
+            Which direction to show the text when annotating.
+        stop_time : float
+            Stop time in seconds
+        color : matplotlib.color | None
+            Matplotlib color
+        """
+
+        import matplotlib.pyplot as plt
+
+        ax = plt.gca()
+        annotation_args = {
+            "va": "top",
+            "y": 0.98,
+            "s": annotation,
+            "transform": ax.get_xaxis_transform(),
+            "color": color,
+        }
+        if not annotation_direction:
+            annotation_direction = "horizontal" if stop_time else "vertical"
+
+        match annotation_direction:
+            case "horizontal":
+                ax.text(
+                    ((start_time + stop_time) / 2 if stop_time else start_time),
+                    ha="center",
+                    rotation=0,
+                    **annotation_args,
+                )
+            case "vertical":
+                if stop_time:
+                    # Prefer using the stop time, as that puts the text _inside_ the shaded area
+                    ax.text(stop_time, ha="right", rotation=90, **annotation_args)
+                else:
+                    ax.text(start_time, ha="right", rotation=90, **annotation_args)
+            case _:
+                raise RuntimeError(
+                    'Invalid value passed for annotation_direction. Expected "horizontal" or '
+                    f'"vertical", got {annotation_direction}.'
+                )
+
+    def highlight_time_range(
+        self,
+        item,
+        annotation=None,
+        annotation_direction=None,
+        color=None,
```

Comment:
be careful with using the same arg name as `mpl`. If you do `channel.highlight_time_range(..., c="g")` you'll get an error
```
TypeError: Got both 'color' and 'c', which are aliases of one another
```

at some point I wrote a little decorator or something to handle this, but I don't think it got merged and I don't remember which feature it was for

Reply:
I think `c` isn't valid for `plt.axvspan`. You end up with `AttributeError: Polygon.set() got an unexpected keyword argument 'c'`.

Reply:
but it is valid for `plt.axvline`. I think it's still good practice not to duplicate/forward mpl args specifically like this if it can be avoided. I've been bitten by this in the past

Reply:
Yeah. I've removed it now.